core/msg_bus.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_msg
+ *
+ * @{
+ *
+ * @file
+ * @brief       Messaging Bus API for inter process message broadcast.
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "irq.h"
+#include "msg_bus.h"
+
+void msg_bus_init(msb_bus_t *bus)
+{
+    static uint16_t bus_count;
+
+    bus->subs.next = NULL;
+    bus->id = bus_count++;
+}
+
+void msg_bus_attach(msb_bus_t *bus, msg_bus_entry_t *entry)
+{
+    unsigned state;
+
+    entry->next.next = NULL;
+    entry->event_mask = 0;
+    entry->pid = sched_active_pid;
+
+    state = irq_disable();
+    list_add(&bus->subs, &entry->next);
+    irq_restore(state);
+}
+
+void msg_bus_detach(msb_bus_t *bus, msg_bus_entry_t *entry)
+{
+    unsigned state;
+
+    state = irq_disable();
+    list_remove(&bus->subs, &entry->next);
+    irq_restore(state);
+}
+
+msg_bus_entry_t* msg_bus_get_entry(msb_bus_t *bus)
+{
```

Comment:
IMO this needs to be guarded with irq_disable/restore.

Reply:
Ok, added.